botocore/client.py

```diff
@@ -898,10 +953,45 @@ def _make_api_call(self, operation_name, api_params):
             'has_streaming_input': operation_model.has_streaming_input,
             'auth_type': operation_model.auth_type,
         }
+
+        if self._ruleset_resolver is None:
+            endpoint_url = self.meta.endpoint_url
+            additional_headers = {}
+        else:
+            endpoint_info = self._ruleset_resolver.construct_endpoint(
+                operation_model=operation_model,
+                call_args=api_params,
+                request_context=request_context,
+            )
+            endpoint_url = endpoint_info.url
+            additional_headers = {
+                # Multi-valued headers are not supported in botocore
+                header_key: header_values[0]
+                for header_key, header_values in endpoint_info.headers.items()
+            }
+            # If authSchemes is present, overwrite default auth type and
+            # signing context derived from service model.
+            auth_schemes = endpoint_info.properties.get('authSchemes')
+            if auth_schemes is not None:
+                (
+                    auth_type,
+                    signing_context,
+                ) = self._ruleset_resolver.auth_schemes_to_signing_context(
+                    auth_schemes
+                )
```

Comment:
This is a case of Black just making a mess. We might be able to amend it with something like:
```suggestion
                auth_info = self._ruleset_resolver.auth_schemes_to_signing_context(
                    auth_schemes
                )
                auth_type, signing_context = auth_info
```